### PR TITLE
Feature/public org members

### PR DESCRIPTION
**Pull request by JVickery-TBS**
### Changes:

This UX feature makes organization and group members more publicly visible.

- Changed <group | organization>/members to extend read_base
- Changed <group | organization>/members route to group_show perms
- Created <group | organization>/manage_members, extending edit_base
- Created /<group_type>/manage_members/<::id> route with group_edit_permissions perms
- Changed redirect for new and delete group/org member to redirect back to manage_members
- Expanded groups_available helper to include dataset and member counts, as well as user context
- Added user context and include member count to organizations_available helper
- Added include member count to group dictization methods
- Added member_count to default group schema with ignore_missing
- Added member counts to the group and org info templates
- Added member counts to the group and orgs list item templates (including the dashboards)
- Added /<::id>/organizations and /<::id>/groups routes to user view
- Added read_groups and read_organizations to user templates (these show what groups and orgs the user belongs to)
- Added group and org count to user secondary content in read_base
- Added various tab links in the templates

### Proposed changes:

May need to generate i18n stuff. Unsure if that is done at this time or not.

### Reason for changes:

The only way to see if a user belongs to a group or organization is to be a sysadmin or admin for the group, then manage the group, and go to members. This is a lot of click-throughs just to view group members, and it is limited to sysadmins and group admins.

User profiles are already visible by default, showing their created datasets. You can click on their created datset and see that it belongs to orgs and groups, thus concluding that the user belongs to said orgs and groups. This is still a lot of click-throughs for non admins to find out if a user belongs to a group.

A user being a member of groups does not seem to be a privacy concern, so making this more easily publicly accessible seems like a fair ux improvement.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [x] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

File: ckan/views/user.py [hunk 1]
```diff
@@ -164,6 +164,54 @@ def read(id: str) -> Union[Response, str]:
...
+    })
+    data_dict: dict[str, Any] = {
+        u'id': id,
+        u'user_obj': current_user,
+        u'include_datasets': False,
+        u'include_num_followers': True
+    }
+    # FIXME: line 331 in multilingual plugins expects facets to be defined.
+    # any ideas?
+    g.fields = []
```
Comment (smotornyuk) [inline ckan/views/user.py:207]: Are you sure you need this?

File: ckan/lib/helpers.py [hunk 2]
```diff
@@ -1986,25 +1989,53 @@ def groups_available(am_member: bool = False) -> list[dict[str, Any]]:
...
     return logic.get_action('group_list_authz')(context, data_dict)
 
 
 @core_helper
 def organizations_available(permission: str = 'manage_group',
-                            include_dataset_count: bool = False
-                            ) -> list[dict[str, Any]]:
+                            include_dataset_count: bool = False,
+                            include_member_count: bool = False,
+                            current_user: ckan.model.User | ckan.model.AnonymousUser | dict = current_user) -> list[dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckan/lib/helpers.py]: accepts username string or None now.

File: ckan/lib/helpers.py [hunk 3]
```diff
@@ -1976,7 +1976,10 @@ def popular(type_: str,
 
 
 @core_helper
-def groups_available(am_member: bool = False) -> list[dict[str, Any]]:
+def groups_available(am_member: bool = False,
+                    include_dataset_count: bool = False,
+                    include_member_count: bool = False,
+                    current_user: ckan.model.User | ckan.model.AnonymousUser | dict = current_user) -> list[dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckan/lib/helpers.py]: only supports `str` type now, does not need `typing.Union`.

File: ckan/lib/helpers.py [hunk 4]
```diff
@@ -1986,25 +1989,53 @@ def groups_available(am_member: bool = False) -> list[dict[str, Any]]:
...
     data_dict = {
         'permission': permission,
-        'include_dataset_count': include_dataset_count}
+        'include_dataset_count': include_dataset_count,
+        'include_member_count': include_member_count}
     return logic.get_action('organization_list_for_user')(context, data_dict)
 
 
+@core_helper
+def member_count(group: ckan.model.Group | dict) -> int:
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckan/lib/helpers.py]: Now accepts group id string.

File: ckan/views/group.py [hunk 5]
```diff
@@ -473,9 +478,17 @@ def about(id: str, group_type: str, is_organization: bool) -> str:
...
         }
     )
-    group_dict = _get_group_dict(id, group_type)
-    group_type = group_dict['type']
-    _setup_template_variables(context, {u'id': id}, group_type=group_type)
+
+    try:
+        group_dict = _get_group_dict(id, group_type)
+        group_type = group_dict['type']
+        _setup_template_variables(context, {u'id': id}, group_type=group_type)
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckan/views/group.py]: `_setup_template_variables` does NOT throw the errors, has been move to right after the try/catch.

File: ckan/lib/helpers.py [hunk 6]
```diff
@@ -1986,25 +1989,53 @@ def groups_available(am_member: bool = False) -> list[dict[str, Any]]:
     :type am-member: bool
 
     '''
-    context: Context = {}
-    data_dict = {'available_only': True, 'am_member': am_member}
+    if type(current_user) is dict:
+        user = current_user['name']
+    else:
+        user = current_user.name
+    context: Context = {'user': user }
```
[the remark this answers is not in the document]
Comment (JVickery-TBS) [reply ckan/lib/helpers.py]: all syntax issues reported by `flake8` have been resolved now.

**Comment by JVickery-TBS** [comment]
@smotornyuk above comments have all been addressed. And I have made all of the copies/updates to the `templates-bs3`.

Changelog file has been added, and syntax issues reported by `flake8` have been resolved. And typing issues from `pyright` have also been addressed.

I have tested out all of the new code changes locally a lot and it behaves as expected.

**Comment by JVickery-TBS** [comment]
@smotornyuk I have added a `member_dump` view for sysadmins and group/org admins to be able to download a CSV of the members with the headers of Username, Email, Name, and Role.

We have had a couple requests over here for this. So figured I would add it in here as well. Let me know if you think it can be improved at all! Thanks!

**Comment by JVickery-TBS** [comment]
@smotornyuk Okay, all tests here should be good to god now.

The circleci one failed at the very end with some `coveralls` issues outside of the tests runs. I do not have the power to re-run the actions however. Would you be able to re-run it to make sure that it is all working? Thanks!

**Comment by smotornyuk** [comment]
@JVickery-TBS, I'm ready to merge it. Can you add to the changelog record a line about renaming `group/organization.members` to `...manage_members` for people, who are using these endpoints in their custom templates?

**Comment by JVickery-TBS** [comment]
@smotornyuk Change log has been updated now

**Pull request merged**